Src/DependencyCollector/Shared/Implementation/TelemetryDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,186 @@
+            this.PrepareExclusionLists(excludeDiagnosticSourceActivities);
+        }
+
+        internal override bool IsSourceEnabled(DiagnosticListener value)
+        {
+            return !this.excludedDiagnosticSources.Contains(value.Name);
+        }
+
+        protected override HashSet<string> GetListenerContext(DiagnosticListener diagnosticListener)
+        {
+            HashSet<string> excludedActivities;
+            if (!this.excludedDiagnosticSourceActivities.TryGetValue(diagnosticListener.Name, out excludedActivities))
+            {
+                return null;
+            }
+
+            return excludedActivities;
+        }
+
+        internal override bool IsEventEnabled(string evnt, object input1, object input2, DiagnosticListener diagnosticListener, HashSet<string> context)
+        {
+            return !this.IsActivityExcluded(evnt, context)
+                && !evnt.EndsWith(ActivityStartNameSuffix, StringComparison.OrdinalIgnoreCase);
+        }
+
+        internal override bool ShouldHandleEvent(KeyValuePair<string, object> evnt, DiagnosticListener diagnosticListener, HashSet<string> context)
+        {
+            return !this.IsActivityExcluded(evnt.Key, context)
+                && evnt.Key.EndsWith(ActivityStopNameSuffix, StringComparison.OrdinalIgnoreCase);
+        }
+
+        internal bool IsActivityExcluded(string activityName, HashSet<string> excludedActivities)
+        {
+            return excludedActivities?.Contains(activityName) == true;
+        }
+
+        internal override void HandleEvent(KeyValuePair<string, object> evnt, DiagnosticListener diagnosticListener, HashSet<string> context)
+        {
+            Activity currentActivity = Activity.Current;
+            if (currentActivity == null)
+            {
+                DependencyCollectorEventSource.Log.CurrentActivityIsNull();
+                return;
+            }
+
+            DependencyCollectorEventSource.Log.TelemetryDiagnosticSourceListenerActivityStopped(currentActivity.Id, currentActivity.OperationName);
+
+            // extensibility point - can chain more telemetry extraction methods here
+            ITelemetry telemetry = this.ExtractDependencyTelemetry(diagnosticListener, currentActivity);
+            if (telemetry == null)
+            {
+                return;
+            }
+
+            // properly fill dependency telemetry operation context
+            telemetry.Context.Operation.Id = currentActivity.RootId;
+            telemetry.Context.Operation.ParentId = currentActivity.ParentId;
+            telemetry.Timestamp = currentActivity.StartTimeUtc;
+
+            foreach (var item in currentActivity.Baggage)
+            {
+                if (!telemetry.Context.Properties.ContainsKey(item.Key))
+                {
+                    telemetry.Context.Properties[item.Key] = item.Value;
+                }
+            }
+
+            this.Client.Initialize(telemetry);
```

Comment:
`Track` calls initialize internally, you don't need to call initialize explicitely

Reply:
good point, thanks. 
should I also remove from ```HttpCoreDiagnosticSourceListener``` or is there a reason for it there?

Reply:
there it's done in Start api and I believe serves a different purpose: it initializes all properties before request headers are injected into an outgoing request.

Reply:
yeah, it seems to make sense there. thanks for explaining. leaving as is.